state-sync/state-sync-v2/data-streaming-service/src/streaming_service.rs

```diff
@@ -272,7 +275,18 @@ impl<T: DiemDataClient + Send + Clone + 'static> DataStreamingService<T> {
 
     /// Returns the data stream associated with the given `data_stream_id`.
     /// Note: this method assumes the caller has already verified the stream exists.
-    fn get_data_stream(&mut self, data_stream_id: &DataStreamId) -> &mut DataStream<T> {
+    fn get_data_stream(&self, data_stream_id: &DataStreamId) -> &DataStream<T> {
+        self.data_streams.get(data_stream_id).unwrap_or_else(|| {
+            panic!(
+                "Expected a data stream with ID: {:?}, but found None!",
+                data_stream_id
+            )
+        })
+    }
+
+    /// Returns the data stream associated with the given `data_stream_id`.
+    /// Note: this method assumes the caller has already verified the stream exists.
+    fn get_data_stream_mut(&mut self, data_stream_id: &DataStreamId) -> &mut DataStream<T> {
```

Comment:
had some borrow checker issues since I need a borrow on the `DataStream` and `&self` (for `streaming_service::notify_bad_response`) : (

this fn is just a copy of the old one, but renamed to `get_data_stream_mut`, since it returns a `&mut DataStream`. I then added `get_data_stream` which looks the exact same but returns `&DataStream`.